src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -51,16 +54,60 @@ public override void OnStopActivity(Activity activity, object payload)
                 return;
             }
 
+            string host;
+
+            if (context.Request.Host.Port is null or 80 or 443)
+            {
+                host = context.Request.Host.Host;
+            }
+            else
+            {
+                host = context.Request.Host.Host + ":" + context.Request.Host.Port;
+            }
+
+            TagList tags;
+
+            // We need following directive as
+            // RouteEndpoint is not available in netstandard2.0 and netstandard2.1
+#if NETCOREAPP
+            var target = (context.GetEndpoint() as RouteEndpoint)?.RoutePattern.RawText;
+
             // TODO: This is just a minimal set of attributes. See the spec for additional attributes:
             // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#http-server
-            var tags = new TagList
+            if (!string.IsNullOrEmpty(target))
+            {
+                tags = new TagList
+                {
+                    { SemanticConventions.AttributeHttpFlavor, context.Request.Protocol },
+                    { SemanticConventions.AttributeHttpScheme, context.Request.Scheme },
+                    { SemanticConventions.AttributeHttpMethod, context.Request.Method },
+                    { SemanticConventions.AttributeHttpHost, host },
+                    { SemanticConventions.AttributeHttpTarget, target },
+                    { SemanticConventions.AttributeHttpStatusCode, context.Response.StatusCode.ToString() },
+                };
+            }
+            else
+            {
+                tags = new TagList
+                {
+                    { SemanticConventions.AttributeHttpFlavor, context.Request.Protocol },
+                    { SemanticConventions.AttributeHttpScheme, context.Request.Scheme },
+                    { SemanticConventions.AttributeHttpMethod, context.Request.Method },
+                    { SemanticConventions.AttributeHttpHost, host },
+                    { SemanticConventions.AttributeHttpStatusCode, context.Response.StatusCode.ToString() },
+                };
+            }
+#else
+            tags = new TagList
             {
-                { SemanticConventions.AttributeHttpMethod, context.Request.Method },
-                { SemanticConventions.AttributeHttpScheme, context.Request.Scheme },
-                { SemanticConventions.AttributeHttpStatusCode, context.Response.StatusCode },
                 { SemanticConventions.AttributeHttpFlavor, context.Request.Protocol },
+                { SemanticConventions.AttributeHttpScheme, context.Request.Scheme },
+                { SemanticConventions.AttributeHttpMethod, context.Request.Method },
+                { SemanticConventions.AttributeHttpHost, host },
+                { SemanticConventions.AttributeHttpStatusCode, context.Response.StatusCode.ToString() },
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3247/files#r866970771 same comment. we can follow up to fix the string alloc.